controllers/dscinitialization/dscinitialization_controller.go

```diff
@@ -104,36 +104,33 @@ func (r *DSCInitializationReconciler) Reconcile(ctx context.Context, req ctrl.Re
 		}
 	}
 
-	if instance.ObjectMeta.DeletionTimestamp.IsZero() {
-		if !controllerutil.ContainsFinalizer(instance, finalizerName) {
-			log.Info("Adding finalizer for DSCInitialization", "name", instance.Name, "finalizer", finalizerName)
-			controllerutil.AddFinalizer(instance, finalizerName)
-			if err := r.Update(ctx, instance); err != nil {
-				return ctrl.Result{}, err
-			}
-		}
-	} else {
+	// manually handle DSCI deletion: this is only to deal with any DSCI creation in the old release with finalizer already added
+	// 1. remove finalizer during DSCI deletion
+	// 2. remove ServiceMesh configurations, it might not have owner as DSCI
+	if !instance.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
proibably this block can be simplified with something like:

```go
if !instance.ObjectMeta.DeletionTimestamp.IsZero() {
    if !controllerutil.RemoveFinalizer(instance, finalizerName) {
	    return  reconcile.Result{}, nil
    }
    
    if err := r.removeServiceMesh(ctx, instance); err != nil {
	    log.Error(err, "Failed to remove ServiceMesh")
	    return reconcile.Result{}, err
    }
    
    if err := r.Update(ctx, instance); err != nil {
	    log.Error(err, "Failed to remove finalizer")
	    return ctrl.Result{}, err
    }
    
    return ctrl.Result{}, nil
}
```

Reply:
hehe , you are  right, i should fast exit the call